### ISSUE TITLE
Compile issues with latest version of MSYS2/GCC (14.1.0)

### ISSUE THREAD

**Issue by ftuffner** [gridlab-d/gridlab-d#1461]
Initial Checklist prior to submitting the issue. Check the bellow boxes to the best of your knowledge:

- [X] I have not found an open issue describing the same problem.
- [X] I went through "CONTRIBUTING.MD" before deciding to submit the issue.
- [X] While building GridLAB-D, I followed all steps as detailed in GridLAB-D Wiki.

------------------------------------------------------------------------------------------------------------

## Please describe what was the cause of the issue?
When compiling with an up-to-date MSYS2 environment, fails very quickly while trying to compile the superLU package with a `ccopy` related error  (see below).

MSYS2 versions:
cmake - 3.29.3
gcc - 14.1.0
cc - 14.1.0

## Please provide the steps to reproduce.
Download GridLAB-D source, use either `release/5.2` or `develop`, and try compiling it.

## Please provide the exact error/warning/failure message printed on the console.
Full log attached, but relevant portion:

```
[17/443] Building C object third_party/superLU_MT/CMakeFiles/superlu_lib.dir/clacon.c.obj
FAILED: third_party/superLU_MT/CMakeFiles/superlu_lib.dir/clacon.c.obj
C:\msys64\mingw64\bin\cc.exe -DHAVE_CONFIG_H -IC:/Code/GLD52ReleaseHELICS/cmake-build/headers -IC:/Code/GLD52ReleaseHELICS/cmake-build/_deps/dlfcn-win32-src/src -D_PTHREAD -DAdd_ -lm -O3 -DNDEBUG -Werror=return-type -Werror=write-strings -static-libgcc -static-libstdc++ -static -w -MD -MT third_party/superLU_MT/CMakeFiles/superlu_lib.dir/clacon.c.obj -MF third_party\superLU_MT\CMakeFiles\superlu_lib.dir\clacon.c.obj.d -o third_party/superLU_MT/CMakeFiles/superlu_lib.dir/clacon.c.obj -c C:/Code/GLD52ReleaseHELICS/third_party/superLU_MT/clacon.c
C:/Code/GLD52ReleaseHELICS/third_party/superLU_MT/clacon.c: In function 'clacon_':
C:/Code/GLD52ReleaseHELICS/third_party/superLU_MT/clacon.c:159:5: error: implicit declaration of function 'ccopy_' [-Wimplicit-function-declaration]
  159 |     ccopy_(&ni, &x[0], &c__1, &v[0], &c__1);
      |     ^~~~~~
[22/443] Building C object _deps/dlfcn-win32-build/src/CMakeFiles/dl.dir/dlfcn.c.obj
ninja: build stopped: subcommand failed.
```

## If possible, please provide any screenshots and other supporting information.
[MSYS2_Compile_errors_20240524.txt](https://github.com/gridlab-d/gridlab-d/files/15438551/MSYS2_Compile_errors_20240524.txt)

-------------------------------------------------------------------------------------------------------------

**Comment by ftuffner** [comment]
As an FYI, the latest version of superLU sequential (6.0.1) compiles without issue under MSYS2.  I've attempted to link it in, but my cmake knowledge isn't the best (despite being a submodule, we don't actually compile the sequential version in).  This might be a prompt to finally replace the "predownloaded" MT version we use (or try updating it).

**Comment by ftuffner** [comment]
Attempted to update to MT 4.0.1 was inconclusive.  The CBLAS package was giving issues that I didn't have a chance to fully debug (compiler-related)

**Comment by ftuffner** [comment]
Above commit is to resolve some VERY slight powerflow differences that gave different starting rotor speeds.  KLU was consistent across platforms, but superLU does slightly different answers on phase A (precision-level differences) - likely due to how MSYS2 vs. Linux handles the BLAS routines/library.

**Comment by ftuffner** [comment]
Resolved as part of PR #1462.  Closing.
